docs/prepare_to_build_docs.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+
+# Build API docs then build docusaurus docs.
+# Currently used in our netlify pipeline.
+
+print_orange () {
+    ORANGE='\033[38;5;208m'
+    NC='\033[0m' # No Color
+    echo -e "${ORANGE}$1${NC}"
+}
+
+print_orange_line () {
+    print_orange "================================================================================"
+}
+
+print_orange_header () {
+    print_orange_line
+    print_orange "$1"
+    print_orange_line
+}
+
+CURRENT_COMMIT=$(git rev-parse HEAD)
+CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+# git pull to get the latest tags
+git pull
+
+print_orange_header "Copying previous versioned docs"
+curl "https://superconductive-public.s3.us-east-2.amazonaws.com/oss_docs_versions_20230404.zip" -o "oss_docs_versions.zip"
+unzip -oq oss_docs_versions.zip -d .
+
+# Move versions.json outside of the repo so there are no conflicts when checking out earlier versions
+VERSIONS_JSON_PATH=../../../versions.json
+mv versions.json $VERSIONS_JSON_PATH
+
+VERSIONS=$(cat $VERSIONS_JSON_PATH | python -c 'import json,sys;obj=json.load(sys.stdin);print(" ".join(obj))')
+
+for version in $VERSIONS; do
+  print_orange_header "Copying code referenced in docs from $version and writing to versioned_code/version-$version"
+
+  git checkout "$version"
```

Comment:
rather than using git to checkout (this screws up your current working branch) can we instead download the zip from github?

Reply:
example zip URL: https://github.com/great-expectations/great_expectations/archive/refs/tags/0.14.13.zip

Reply:
Good call - I was looking for a way to download directly from github earlier, thank you! I'll edit 🙇

Reply:
Fixed and simplified a few other steps as well since we are on the same branch (except when building api docs from develop where we use the most recent released version) 🙇